app/src/main/java/com/mdg/droiders/samagra/shush/adapters/TimeListAdapter.java

```diff
@@ -0,0 +1,495 @@
+        refreshCursor();
+        alarmScheduler.setWeeklyAlarm(startTime, endTime, getDayArr(holder), holder.id);
+    }
+
+    /**
+     * Updates db and sets alarm for the corresponding day.
+     *
+     * @param holder The holder instance that is currently bound to the row whose
+     *               alarm is set.
+     * @param day    The day for which alarm is set. The day is zero indexed and starts from monday.
+     */
+    private void notifyDayAlarmSet(TimeRowHolder holder, int day) {
+
+        ContentValues values = new ContentValues();
+        values.put(getColumnFromDay(day), true);
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        alarmScheduler.setSingleDayAlarm(startTime, endTime, day, holder.id);
+    }
+
+    /**
+     * Updates db and cancels alarm for the corresponding day.
+     *
+     * @param holder The holder instance that is currently bound to the row whose
+     *               alarm is cancelled.
+     * @param day    The day for which alarm is cancelled. The day is zero indexed and starts from monday.
+     */
+    private void notifyDayAlarmCancelled(TimeRowHolder holder, int day) {
+
+        ContentValues values = new ContentValues();
+        values.put(getColumnFromDay(day), false);
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        alarmScheduler.cancelSingleAlarm(day, holder.id);
+    }
+
+    /**
+     * Returns the corresponding column of the day in db.
+     *
+     * @param day The day for which the column is to be returned
+     * @return The column corresponding to the given day
+     */
+    private String getColumnFromDay(int day) {
+        switch (day) {
+            case 0: {
+                return PlacesContract.TimeEntry.COLUMN_MONDAY;
+            }
+            case 1: {
+                return PlacesContract.TimeEntry.COLUMN_TUESDAY;
+            }
+            case 2: {
+                return PlacesContract.TimeEntry.COLUMN_WEDNESDAY;
+            }
+            case 3: {
+                return PlacesContract.TimeEntry.COLUMN_THURSDAY;
+            }
+            case 4: {
+                return PlacesContract.TimeEntry.COLUMN_FRIDAY;
+            }
+            case 5: {
+                return PlacesContract.TimeEntry.COLUMN_SATURDAY;
+            }
+            case 6: {
+                return PlacesContract.TimeEntry.COLUMN_SUNDAY;
```

Comment:
Things like this `case` statement would be just three lines, instead of current 20+ lines and would decrease writing and testing time. Testing because currently, you have 7 possible points of failure instead of 1.

Reply:
I have used an array of checkboxes now, I will push the code soon. It did reduce many lines of code 😄